x-pack/plugins/apm/public/components/routing/templates/apm_service_template.tsx

```diff
@@ -127,22 +124,29 @@ function AnalyzeDataButton({ serviceName }: { serviceName: string }) {
   const { rangeTo, rangeFrom, environment } = urlParams;
   const basepath = services.http?.basePath.get();
 
-  if (isRumAgentName(agentName) || isIosAgentName(agentName)) {
+  if (
+    (isRumAgentName(agentName) || isIosAgentName(agentName)) &&
+    rangeFrom &&
+    rangeTo
+  ) {
     const href = createExploratoryViewUrl(
       {
-        'apm-series': {
-          dataType: isRumAgentName(agentName) ? 'ux' : 'mobile',
-          time: { from: rangeFrom, to: rangeTo },
-          reportType: 'kpi-over-time',
-          reportDefinitions: {
-            [SERVICE_NAME]: [serviceName],
-            ...(!!environment && ENVIRONMENT_NOT_DEFINED.value !== environment
-              ? { [SERVICE_ENVIRONMENT]: [environment] }
-              : {}),
+        reportType: 'kpi-over-time',
+        allSeries: [
+          {
+            order: 0,
+            name: `${serviceName}-series`,
+            dataType: isRumAgentName(agentName) ? 'ux' : 'mobile',
+            time: { from: rangeFrom, to: rangeTo },
+            reportDefinitions: {
+              [SERVICE_NAME]: [serviceName],
+              ...(!!environment && ENVIRONMENT_NOT_DEFINED.value !== environment
```

Comment:
How does this behave with `ENVIRONMENT_ALL`? It looks like this isn't actually changing, but would you like to go ahead and fix #105467 while you're in here?